jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/codegeneration/MethodCall.java

```diff
@@ -393,6 +403,35 @@ private Method findMethodInClassHierarchyMatchingParams(String methodName, Class
     return ret;
   }
 
+  private boolean shouldSkipChaining(MethodParam... params) {
+    return params.length > 0 && Arrays.stream(params).allMatch(MethodParam::isDefault);
+  }
+
+  private MethodParam[] adjustParamsForBoolParam(String methodName, MethodParam[] params) {
+    if (params.length == 1 && params[0] instanceof BoolParam) {
+      Method method = findMethodInClassHierarchyMatchingParams(
+              methodName, returnType, new MethodParam[0]);
+      if (method != null) {
+        return new MethodParam[0];
+      }
+    }
+    return params;
+  }
+
+  private Method findMethodForChaining(String methodName, MethodParam[] params) {
+    return findMethodInClassHierarchyMatchingParams(methodName, returnType, params);
+  }
+
+  private UnsupportedOperationException noMatchingMethodFoundException(
```

Comment:
Any particular reason for creating a separate method and changing the exception message? I think the original message is more useful for the final user (if it ever arises, which would indicate a problem with jmeter-java-dsl code, and not something the user did wrong).